addons/udes_delivery_control/models/stock_picking.py

```diff
@@ -166,6 +171,21 @@ def _compute_show_mark_as_todo(self):
         for picking in self.filtered("u_is_delivery_control"):
             picking.show_mark_as_todo = picking.state == "draft"
 
+    @api.onchange("u_loading_type")
+    def _compute_loading_type(self):
```

Comment:
Why is this method in both `stock.picking` and `stock.picking.extras`? I might be missing something but in the picking can't we just take the value from the extras record?

Reply:
The onchange on loading_type field will hide/unhide fields on delivery control picking type. We can't fetch value from `stock.picking.extras` because at this point won't have been updated on the related field ( the write back doesn't happened until we save the record)

Reply:
In that case could we refactor out the logic into a `get_loading_type` method and call this from both compute methods? Not sure if that's practical, but would be good to avoid duplicated code if possible. Will have a think and get back to you.

Reply:
Resolving as no longer relevant (both entities use different field names).